rasa/shared/core/events.py

```diff
@@ -458,6 +458,10 @@ def __init__(
         }
 
         if parse_data:
+            if "text_tokens" in parse_data:
+                self.parse_data["text_tokens"] = [
+                    (t.start, t.end, t.text) for t in parse_data.pop("text_tokens")
```

Comment:
should we use a dict instead of a tuple to make the single parts easier identifiable? Cause `json.dumps` will make the tuple a list and then you might up with something like `[1,2, "3"]` which is not exactly intuitive (the list is more memory efficient though)

Reply:
We also don't necessarily need the actual test of the token. It might be worth asking Rasa X folks if they will need it.

Reply:
Ok sure, I'll ask if they need the actual text.
What would the dict look like in both cases though:
- with text: `{t.text: [t.start, t.end]}`
- without text: `{t.start: t.end}`?

Reply:
mhm, I personally don't like dictionaries where I don't know the keys (unless I'd always just iterate over the entire thing anyway) because they are super awkward to access. I'd probably even be verbose. If they don't need the text I'd go for the tuple (that's clear enough in that case I think) (`t.start, t.end`)

Reply:
Yeah I'd say either tuple e.g. `(1,5)` or if we need text then a full dict e.g. `{"start": 1, "end": 5, "text" "hello"}`

Reply:
To follow-up: Practice CDD confirmed they don't need the text, so I kept the tuple type `(t.start, t.end)`.